File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    answer = 0
    storage = []
    j = 0
    for i in range(1, len(order) + 1):
        storage.append(i)
        while storage and (storage[-1] == order[j]):
            answer += 1
            j += 1
            storage.pop()

    return answer

File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
    let storage = [], n = 1, answer = 0
    
    for (let i = 0; i < order.length; i++){
        let o = order[i]
        if (storage.length && storage[storage.length-1] == o){
            storage.pop()
            answer++
            continue
        }
        
        while (o != n){
            storage.push(n)
            n++
            if (n > order.length) return answer
        }
        n++
        answer++
    }
    
    return answer
}